superset/security/manager.py

```diff
@@ -2401,6 +2428,32 @@ def raise_for_access(  # noqa: C901
                             and slc in dashboard_.slices
                             and slc.datasource == datasource
                         )
+                        or (
```

Comment:
Oh my, what is this monster expression stating on line 2379?!? 😱

We should break this down into methods that are self-explanatotry, like:

```python
return (
    self.is_d2d_allowed(form_data) or
    self.is_dashbaord_rbac(form_data) or
    ...
)
```

(Not saying _you_ should do it, but you could start by moving the code you're adding into a method and call that instead here.)

Reply:
Great idea! Will do with this new block and then open a dedicated follow up later